Implement Unit tests for DBClient lib

gportog opened the pull request:
- [x] DBClient constructor
- [x] DBClient.get
- [x] DBClient.getById
- [x] DBClient.insert
- [x] DBClient.update
- [x] DBClient.delete

Signed-off-by: Gustavo Porto Guedes <gustavo.guedes@fatec.sp.gov.br>

gabrieldrs commented on spec/DBClientSpec.js:
```diff
@@ -0,0 +1,276 @@
+                .toThrowError('This client has no default databaset set, and one was not provided.');
+        });
+
+        it('should return an array of valid menus schema when initiating the db menus on the constructor', async () => {
```
`async ()` is not working on my end, not sure if this is a valid syntax

gportog committed:
Implement Unit tests for DBClient lib

- [x] DBClient constructor
- [x] DBClient.get
- [x] DBClient.getById
- [x] DBClient.insert
- [x] DBClient.update
- [x] DBClient.delete

Signed-off-by: Gustavo Porto Guedes <gustavo.guedes@fatec.sp.gov.br>

gabrieldrs commented on spec/DBClientSpec.js:
```diff
@@ -0,0 +1,253 @@
+            let isString = true;
+            let dbInstance = new DBClient('menus-test');
+            if (typeof dbInstance._db !== 'string') isString = false;
+            expect(isString).toBe(true);
```
in this case you could also use `toBeTruthy()`, just keep in mind that `toBeTruthy()` is more generic than `toBe(value)`, because things like "a", 1.0, 42, etc are considered "Truthy".

gabrieldrs replied:
This is not a blocker though, just a comment

gportog replied:
OK, thanks! :+1:

gabrieldrs commented on spec/DBClientSpec.js:
```diff
@@ -0,0 +1,253 @@
+            let menus = await dbInstance.get({ schemaName: 'menus' });
+            let checkMenus = await jsonValidator.sanitizeList(menus, 'menus');
+            if (typeof checkMenus === 'object') menusType = 'object'
+            expect(menusType).toEqual('object');
```
throughout your code I noticed you repeat this structure many times, in which you first do a comparison in order to set a variable to a value that you either expect it to be equal or different a specific value. Instead of doing that you could transform the if statement into an actual `expect`. For this test you could replace lines 67, 68 with:

```
expect(typeof checkMenus).toBe('object');
```

gabrieldrs replied:
Please update your expects so it's not necessary to run the if statement

gportog replied:
Got it! :+1:

gabrieldrs commented on spec/DBClientSpec.js:
```diff
@@ -0,0 +1,253 @@
+            } catch (err) {
+                errMessage = err.message;
+            }
+            expect(errMessage).toEqual('Database does not exist.');
```
could you please try to use the same structure used in line 58 here?

gabrieldrs replied:
and in the following try catch blocks

gabrieldrs commented on spec/DBClientSpec.js:
```diff
@@ -0,0 +1,253 @@
+            let dbInstance = new DBClient();
+            let jsonValidator = new JSONValidator();
+            let menus = await dbInstance.get({ db: 'menus-test', schemaName: 'menus' });
+            let checkMenus = await jsonValidator.sanitizeList(menus, 'menus');
```
this is not recommended because if we have a bug in our json validator, this could be unreliable.

If we want to test that the schema returned matches a required schema, we must use our validator directly here (ajv)

gabrieldrs reviewed (approved):
Such a great use of async and await in these tests, huge improvements, congrats!!!
